airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/format/containers.py

```diff
@@ -95,41 +93,47 @@ def format_java_container(dagger_client: dagger.Client) -> dagger.Container:
             " spotless:check"
             " clean"
         ],
-        include=["**/*.java", "**/*.gradle"],
+        dir_to_format=java_code,
     )
 
 
-def format_js_container(dagger_client: dagger.Client) -> dagger.Container:
-    """Format yaml and json code via prettier."""
+def format_js_container(dagger_client: dagger.Client, js_code: dagger.Directory) -> dagger.Container:
+    """Create a Node container with prettier installed with mounted code to format and a cache volume."""
     return build_container(
         dagger_client,
         base_image=NODE_IMAGE,
-        include=["**/*.yaml", "**/*.yml", "**.*/json", "package.json", "package-lock.json"],
+        dir_to_format=js_code,
         install_commands=["npm install -g npm@10.1.0 prettier@3.0.3"],
+        cache_volume=dagger_client.cache_volume("js"),
```

Comment:
This cache volume combined with changing the prettier command to 
```
[f"prettier --write . --list-different --cache --cache-location={CACHE_MOUNT_PATH}/.prettier_cache"],
```

leads to a significant boost: 8s run vs 30s duration on subsequent runs.

Reply:
cf https://github.com/airbytehq/airbyte/pull/33097#issuecomment-1845438164